src/executables/csql.c

```diff
@@ -1460,6 +1464,205 @@ csql_do_session_cmd (char *line_read, CSQL_ARGUMENT * csql_arg)
 	}
 
       break;
+
+    case S_CMD_CONNECT:
+      /*dbname_cpy can be stored DB name+ @ + remote hostname + \0 */
+      char dbname_cpy[MAXLOGNAME + CUB_MAXHOSTNAMELEN + 2];
+      char username_cpy[DB_MAX_USER_LENGTH + 1];
+      char host_cpy[DB_MAX_USER_LENGTH + 1];
+      const char *delim = " @\n";
+      char *save_ptr_strtok = NULL;
+      char *db_name = NULL;
+      char *user_name = NULL;
+      char *host_name = NULL;
+      char *p = NULL;
+      CSQL_ARGUMENT csql_new_arg;
+
+      /*find @hostname */
+      host_name = strchr (argument, '@');
+      if (host_name != NULL)
```

Comment:
How about this code?

```
if (host_name != NULL)
  {
    if (*(host_name + 1) == '\0')
      {
        host_name = NULL;
        csql_Error_code = CSQL_ERR_CONNECT;
        return DO_CMD_FAILURE;
      }

      if (strcasecmp ("127.0.0.1", host_name +1 ) == 0 || strcasecmp ("localhost", host_name +1) == 0
            || strcasecmp(boot_get_host_name(), host_name+1) == 0 || strcasecmp(boot_get_ip(), host_name+1) == 0)
        {
          *(host_name ) = '\0';
        }
    }

```

Reply:
1. It keeps the unity with 1520 line `db_name += 1;`
2. Could you explain why you suggest this one?

Reply:
To me mine is better to read. and even unity because demo version style was like yours but it is so hard to read.